### DISCUSSION TITLE
Difference between bentoml.Service(), and @bentoml.service

### DISCUSSION THREAD

**Discussion by abi-jey** [bentoml/BentoML#4504] [General]
What is the difference between creating service via 
1) bentoml.Service() then adding api routes like @service.api 
2) creating class with decorator @bentoml.service

The reason that I'm asking question is that I want to use class decorator(Option 2) to use a config, but the same doesn't seem to be possible with (Option 1)

**Comment by parano** [comment]
@realabja 

- `bentoml.Service` is a legacy API from 1.0.x and 1.1.x releases. It is still available in the latest 1.2.x versions for backwards compatibility.
- `@bentoml.service` decorator is the recommended API if you're getting started with BentoML today.
